### PR TITLE
(#1093) show verbose errors only in dev env

### PR DESCRIPTION
**Pull request by atolopko-czi**
addresses security vulnerability on prod/staging

### Reviewers
**Functional:** 
@MDunitz 

**Readability:** 
@blrnw3 

---

## Changes
only enable debug logging and and flask debug mode when in dev mode (more precisely, when not in "prod" or "stage" mode).

## Definition of Done (from ticket)
N/A.

## QA steps (optional)
Ran local stack and manually smoke tested data portal to ensure server came up.
But not sure how to re-test the specific error generated during the security audit.

### CODE REVIEW

File: backend/chalice/api_server/app.py [hunk 1]
```diff
@@ -31,14 +31,17 @@ def create_flask_app():
...
-    flask_app.debug = True
-    app.debug = flask_app.debug
-    app.log.setLevel(logging.DEBUG)
 
     # set the flask secret key, needed for session cookies
     flask_secret_key = "OpenSesame"
     deployment = os.environ["DEPLOYMENT_STAGE"]
     allowed_origins = []
+
+    if deployment not in ["prod", "stage"]:
```
Comment (atolopko-czi) [inline backend/chalice/api_server/app.py:40]: not sure if it matters for staging env, but sticking with requirement per the issue.

**Comment by blrnw3** [comment]
Is chalice even used any more? I thought that was removed in favor of using gunicorn on ECS. See `backend/corpora/api_server/app.py`. In that file it seems the debug flag is correctly removed in prod

**Comment by MDunitz** [comment]
> Is chalice even used any more? I thought that was removed in favor of using gunicorn on ECS. See `backend/corpora/api_server/app.py`. In that file it seems the debug flag is correctly removed in prod



> Is chalice even used any more? I thought that was removed in favor of using gunicorn on ECS. See `backend/corpora/api_server/app.py`. In that file it seems the debug flag is correctly removed in prod

Chalice should no longer be used. I'm going to try removing it and deploying to rdev

**Comment by atolopko-czi** [comment]
> > Is chalice even used any more? I thought that was removed in favor of using gunicorn on ECS. See `backend/corpora/api_server/app.py`. In that file it seems the debug flag is correctly removed in prod
> 
> > Is chalice even used any more? I thought that was removed in favor of using gunicorn on ECS. See `backend/corpora/api_server/app.py`. In that file it seems the debug flag is correctly removed in prod
> 
> Chalice should no longer be used. I'm going to try removing it and deploying to rdev

Closing this PR, but not the story! Will look into where else we might need to update the logging. Presumably the bug description was wrong in identifying the location of the required configuration change.

**Pull request closed without merging**